src/hotspot/os/linux/os_linux.cpp

```diff
@@ -3343,14 +3342,14 @@ bool os::committed_in_range(address start, size_t size, address& committed_start
 
   committed_start = nullptr;
 
-  int loops = (pages + stripe - 1) / stripe;
+  int loops = checked_cast<int>((pages + stripe - 1) / stripe);
   int committed_pages = 0;
   address loop_base = start;
   bool found_range = false;
 
   for (int index = 0; index < loops && !found_range; index ++) {
     assert(pages > 0, "Nothing to do");
-    int pages_to_query = (pages >= stripe) ? stripe : pages;
+    size_t pages_to_query = (pages >= stripe) ? stripe : pages;
```

Comment:
Pre-existing: Could you instead make `pages` and `pages_to_query` and below `vecIdx` an uintx? Should be the same and give you no new errors, but size_t as "number of things" is not right.

Reply:
This seems to not complain.

Reply:
/home/runner/work/jdk/jdk/src/hotspot/os/linux/os_linux.cpp:3352:34: error: comparison of integer expressions of different signedness: 'intx' {aka 'long int'} and 'const size_t' {aka 'const long unsigned int'} [-Werror=sign-compare]
 3352 |     intx pages_to_query = (pages >= stripe) ? stripe : pages;
      |                            ~~~~~~^~~~~~~~~